activerecord/lib/active_record/autosave_association.rb

```diff
@@ -417,7 +417,8 @@ def save_has_one_association(reflection)
                 record[reflection.foreign_key] = key
               end
 
-              saved = record.save(:validate => !autosave)
+              _validate = !autosave && validate
```

Comment:
This method is defined in `Validations` module and it's not related with the current transaction. I was working on `ActiveRecord 4.2.3` and somehow couldn't transfer all my changes to this pr, thank you soo much for this nice catch!
We have to pass `validate` option to autosave_association callbacks I'm currently working on it.